pathway-analysis/ssgsea_example.Rmd

```diff
@@ -0,0 +1,392 @@
+
+```{r}
+dim(ssgsea_results)
+```
+
+```{r}
+head(rownames(ssgsea_results))
+```
+
+#### Identifying "interesting" features
+
+One way to figure out what pathways may be interesting is to find which pathways
+have the highest _variance_ in their ssGSEA scores.
+We can calculate the row variances with the `matrixStats` package.
+
+```{r}
+score_row_var <- matrixStats::rowVars(ssgsea_results)
+# for convenience, let's use the pathway names with this vector
+names(score_row_var) <- rownames(ssgsea_results)
+# look at the 'top' pathways by this metric
+head(sort(score_row_var, decreasing = TRUE))
+```
+
+Note that many of these pathways seem to have something to do with the immune
+system and may very well share a large number of genes.
+Let's look at an example pair of these pathways.
+
+```{r}
+intersect(kegg_pathways$KEGG_ALLOGRAFT_REJECTION, 
+          kegg_pathways$KEGG_GRAFT_VERSUS_HOST_DISEASE)
+```
+
+These ssGSEA scores should not be treated as if they are independent.
+
+#### Plotting features
+
+What do the patterns across strains and tissues look like?
+Let's look at the KEGG Complement and Coagulation Cascades pathway as an 
+example.
+
+We'll be using `ggplot2` for plotting, so we'll want our ssGSEA values as a
+`data.frame`.
+
+```{r}
+comp_coag_df <- data.frame(
+  sample_accession = colnames(ssgsea_results),
+  ssgsea_score = ssgsea_results["KEGG_COMPLEMENT_AND_COAGULATION_CASCADES", ]
+)
+```
+
+We need to add in the strain, tissue, and condition metadata to make an 
+informative plot.
+
+```{r}
+comp_coag_df <- metadata_df %>%
+  dplyr::inner_join(comp_coag_df, 
+                    by = c("refinebio_accession_code" = "sample_accession"))
+```
+
+```{r}
+comp_coag_df %>% 
+  ggplot2::ggplot(ggplot2::aes(x = condition, 
+                               y = ssgsea_score)) +
+  ggplot2::geom_boxplot() +
+  ggplot2::facet_grid(strain ~ refinebio_specimen_part) +
+  ggplot2::labs(y = "KEGG COMPLEMENT AND COAGULATION CASCADES") +
+  # x-axis text at a 45 degree angle to increase readability
+  ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 45, hjust = 1))
+```
+
+```{r}
+# saving to file
+ggplot2::ggsave(file.path("plots", "GSE75574_ssgsea_comp_coag_facet.pdf"))
+```
+
+Looks like there may be some differences between tissues in this pathway, with
+some strains showing more pronounced differences in white adipose tissue in the 
+calorie-restricted vs. control comparison.
+We should always follow up with more formal analysis (e.g., QuSAGE), but 
+this series of steps is a good way to explore one's data.
+
+#### Aside: Gene set _size_ influences ssGSEA score values
+
+ssGSEA score values are not necessarily comparable between gene sets of 
+different sizes (number of genes).
+If you are comparing ssGSEA scores between samples _within a gene set_, this is
+not a concern.
+
+To demonstrate the effect of gene set size on scores, we'll perform a short 
+experiment with random gene sets.
+
+```{r}
```

Comment:
This is a pretty long code chunk. Do you want to break it up a bit? Maybe turn some of your comments into text?

Reply:
I kept this altogether because not everyone will necessarily want to go through it.